### PR TITLE
Add space sorting options: `size` and `mdate`

### PR DESCRIPTION
**Pull request by JammingBen**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/8439
- Fixes https://github.com/owncloud/web/issues/8409

## Screenshots (if appropriate):

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

File: packages/web-app-files/src/composables/sort/useSort.ts [hunk 1]
```diff
@@ -63,8 +65,13 @@ export function useSort<T extends SortableItem>(options: SortOptions<T>): SortRe
...
   const handleSort = ({ sortBy, sortDir }: { sortBy: string; sortDir: SortDir }) => {
-    sortByRef.value = sortBy
-    sortDirRef.value = sortDir
+    return router.replace({
+      query: {
+        ...unref(router.currentRoute).query,
+        'sort-by': sortBy,
+        'sort-dir': sortDir
+      }
+    })
```
Comment (JammingBen) [inline packages/web-app-files/src/composables/sort/useSort.ts:74]: This fixes https://github.com/owncloud/web/issues/8409.

We (currently) need to update those params at the same time, otherwise one of the 2 changes will get swallowed by the other. This happens because of the new async behaviour of `vue-router`.

Another, more global approach to fix this can be found here: https://github.com/owncloud/web/pull/8414. But it has other drawbacks, hence I figured we solve it this way - at least for now.

Comment (dschmidt) [reply packages/web-app-files/src/composables/sort/useSort.ts:74]: Maybe worth putting this comment into source?

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 2]
```diff
@@ -51,13 +51,15 @@
                 v-oc-tooltip="showSpaceMemberLabel"
                 :aria-label="showSpaceMemberLabel"
                 appearance="raw"
-                @click="openSidebarSharePanel(resource)"
+                @click="openSidebarSharePanel(resource as SpaceResource)"
```
Comment (dschmidt) [inline packages/web-app-files/src/views/spaces/Projects.vue:54]: Would be nice to figure out how to type slots :-D

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 3]
```diff
@@ -51,13 +51,15 @@
...
               >
                 <oc-icon name="group" fill-type="line" />
               </oc-button>
             </template>
             <template #contextMenuActions="{ resource }">
-              <space-context-actions :action-options="{ resources: [resource] }" />
+              <space-context-actions
+                :action-options="{ resources: [resource] as SpaceResource[] }"
+              />
```
Comment (dschmidt) [inline packages/web-app-files/src/views/spaces/Projects.vue:62]: I don't disagree, just curious: why was this necessary?

**Review by dschmidt — APPROVED** [review]
LGTM

File: packages/web-app-files/src/views/spaces/Projects.vue (hunk 3 above)
Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/Projects.vue:62]: My IDE complains that the types are incompatible: `Type Resource[] is not assignable to type SpaceResource[]`

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/34010/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/34010/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2023-3-22-02-48-35.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/34010/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2023-3-22-02-49-03.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/34010/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2023-3-22-02-49-09.zip 


</p></details>
:boom: The e2e-oc10 tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/views/spaces/Projects.vue (hunk 2 above)
Comment (dschmidt) [reply packages/web-app-files/src/views/spaces/Projects.vue:54]: Turns out IntelliJ is smart enough to deduce the type from the slot declaration, vue-tsc apparently isn't... anyhow this is the correct fix, as resource-tiles component  just deals with resources and that's why we need to downcast here. 

Would be great if Vue had support for generic components 👀

File: packages/web-app-files/src/views/spaces/Projects.vue (hunk 3 above)
Comment (dschmidt) [reply packages/web-app-files/src/views/spaces/Projects.vue:62]: Yep, as discussed: makes sense and is correct (and so is this change)

**Review by dschmidt — APPROVED** [review]

File: packages/web-app-files/src/composables/sort/useSort.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { ref, Ref, computed, unref, isRef } from 'vue'
 import { MaybeRef, ReadOnlyRef } from 'web-pkg/src/utils'
 import { useRouteName, useRouteQueryPersisted, QueryValue } from 'web-pkg/src/composables'
 import { SortConstants } from './constants'
+import { useRouter } from 'vue-router'
```
[the remark this answers is not in the document]
Comment (kulmann) [reply packages/web-app-files/src/composables/sort/useSort.ts:5]: out of curiosity, what kind of problems?

File: packages/web-app-files/src/views/spaces/Projects.vue (hunk 2 above)
Comment (kulmann) [reply packages/web-app-files/src/views/spaces/Projects.vue:54]: haha +1 for generic components 😅

File: packages/web-app-files/src/composables/sort/useSort.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dschmidt) [reply packages/web-app-files/src/composables/sort/useSort.ts:5]: Not returning our router instance or even: no instance at all.

As I said I haven't debugged it, so I can't tell you what exactly is going on.
